### PR TITLE
Sprint 42

### PR DESCRIPTION
**Pull request by ericsorenson**
# Sprint 42

### CODE REVIEW

**Comment by kfoley-18F** [comment]
We use this list when performing a code review to ensure that all tasks have been completed.

- [ ] begin by deleting any existing copy of the Git repository, running `docker system prune` to remove any cached artifacts from any prior code review
- [ ] clone a fresh copy of the Git repository (`git clone https://github.com/ustaxcourt/ef-cms.git`), fetch the pull request and name it for the sprint (e.g., `git fetch origin pull/204/head:sprint-43`), and then switch to that branch (e.g. `git checkout sprint-43`)
- [ ] review the pull request itself, to get oriented
	- [ ] read the description of the pull request, which should summarize the changes made
	- [ ] read through every task on the Scrum board that's encompassed by this pull request
	- [ ] read the description of the commits that comprise the pull request
- [ ] stand up the site locally, with `./docker-run.sh`
	- [ ] test all functionality in all major browsers, emphasizing the functionality that this pull request addresses
		- [ ] for internal Court functionality, perform the most thorough testing in Chrome, though also test in Edge and Firefox
		- [ ] for public-facing functionality, test in browsers consistent with [public browser use data](https://analytics.usa.gov/), including Internet Explorer 11
		- [ ] test in Mobile Safari and Mobile Chrome, with the caveat that not all internal Court functionality will be necessary on these platforms
	- [ ] use an automated audit tool for code quality and practices (recommended: [Chrome DevTools](https://developers.google.com/web/tools/chrome-devtools/), aka [Lighthouse](https://chrome.google.com/webstore/detail/lighthouse/blipmdconlkpinefehnmjammfjpmpbjk))
		- [ ] look at efficiency of page loads, asset sizes, HTTP connection management, etc.
	- [ ] review for accessibility
		- [ ] use an automated audit tool, such as Chrome Audit or aXe
		- [ ] navigate site only with the keyboard
		- [ ] use VoiceOver or Narrator to navigate the site with audio only, with the display turned off
		- [ ] manually test anything that pa11y cannot test automatically (e.g., contrast of text over images)
- [ ] [review static code analysis results](https://sonarcloud.io/organizations/flexion-github/projects) in the vendor’s SonarCloud environment
- [ ] examine OWASP ZAP output in `docs/`, to ensure that any errors are known to be false positives or have been previously declared to be acceptable
- [ ] skim all new code, in the context of existing code, [looking for problems](#what-we-look-for) (knowing that the vast majority of new code will be covered by tests)
- [ ] review all tests
	- [ ] look at code coverage of tests in [the vendor's SonarCloud environment](https://sonarcloud.io/organizations/flexion-github/projects)
	- [ ] methodically review all new tests for correctness, quality of naming
- [ ] determine what code isn’t tested, review that rigorously
- [ ] review documentation to ensure that it matches changes
- [ ] provide comments on the pull request on GitHub, as necessary
	- [ ] for comments that are specific to a particular line of code, comment on those specific lines
	- [ ] for comments that are more general, attach the comment to a random line in `README.md` (as opposed to commenting on the pull request itself), to be able to use GitHub's ability to thread discussions on those comments
- [ ] for each feature-level bug (i.e., it’s working as designed, but designed wrong), open a new issue and put it in the backlog

File: docs/entities/IrsPractitioner.md [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+                      limit: 100
+              otherwise: 
+                type: "string"
+                flags: 
+                  presence: "required"
+                rules: 
+                  - 
+                    name: "pattern"
+                    args: 
+                      regex: "/^(\\d{5}|\\d{5}-\\d{4})$/"
```
Comment (kfoley-18F) [inline docs/entities/IrsPractitioner.md:133]: Same here - can y'all please add a comment/descriptor to the regex? thanks!

File: shared/src/business/entities/EntityConstants.js [hunk 2]
```diff
@@ -0,0 +1,779 @@
...
+const documentMapInternal = require('../../tools/internalFilingEvents.json');
+const { sortBy } = require('lodash');
+
+const SERVICE_INDICATOR_TYPES = {
+  SI_ELECTRONIC: 'Electronic',
+  SI_NONE: 'None',
+  SI_PAPER: 'Paper',
+};
+
+const DOCKET_NUMBER_MATCHER = /^([1-9]\d{2,4}-\d{2})$/;
```
Comment (kfoley-18F) [inline shared/src/business/entities/EntityConstants.js:12]: Same here please! Short comment on what the regex is supposed to filter - and/or examples.

File: shared/src/business/entities/EntityConstants.js [hunk 3]
```diff
@@ -0,0 +1,779 @@
+const courtIssuedEventCodes = require('../../tools/courtIssuedEventCodes.json');
```
Comment (kfoley-18F) [inline shared/src/business/entities/EntityConstants.js:1]: This is great - love to see these consolidated.

File: docs/entities/Case.md [hunk 4]
```diff
@@ -259,7 +259,7 @@
         - 
           name: "pattern"
           args: 
-            regex: "/^(\\d{3,5}-\\d{2})$/"
+            regex: "/^([1-9]\\d{2,4}-\\d{2})$/"
```
[the remark this answers is not in the document]
Comment (kfoley-18F) [reply docs/entities/Case.md:262]: https://nasamuffin.github.io/regex/documentation/2019/05/08/documenting-regex.html
might be a reasonable starting place for discussion, but there's flexibility here. 

I also noticed that the regex seems to be copied across several markdown locations. Maybe there's also opportunity for some cross-linking of definitions? 🤷

**Comment by kfoley-18F** [comment]
@ericsorenson - I also found a runtime issue and reported in slack: 
https://ustaxcourt.slack.com/archives/CD4PNKEPK/p1592612336220500

**Comment by kfoley-18F** [comment]
@JessicaMarine - I recommend the Court accept this work into Staging.

File: docs/entities/Case.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (ericsorenson) [reply docs/entities/Case.md:262]: These markdown files are all autogenerated from JS entities. I'll add more documentation in the constants file where the regex values originate.

**Pull request merged**